### DISCUSSION TITLE
Can Next.js be hosted on GoDaddy?

### DISCUSSION THREAD

**Discussion by solutionstack99** [vercel/community#7433] [Help]
### Summary

I built a website with Next.js + Prismic for a client and I was planning to host it with Vercel but my client wanted to host it with GoDaddy.

As I know there will be another feature that builds with Machine learning for a dashboard when a user login which I didn't build it.

My questions are:

1. Is it practical to host it with GoDaddy where they use CPanel?
2. How to set up Webhook if they use GoDaddy to trigger real-time updates?
3. Is there anything else I should be aware of regarding the PROS and CONS of deploying Prismic beside API routes?

Thank you in advance!

### Steps to Reproduce

No steps to reproduce.

**Comment by pawlean** [comment]
Hi @solutionstack99!

I've not done this before but after some research, this is what I found.

> Is it practical to host it with GoDaddy where they use CPanel?

Hosting a Next.js application on GoDaddy’s shared hosting is impractical due to limited Node.js support, potential performance bottlenecks, and scalability issues inherent in shared environments. GoDaddy’s shared hosting is optimized for PHP applications rather than Node.js, which is needed for Next.js.

Hosting on Vercel or Netlify are recommended as they are designed specifically for modern JavaScript frameworks. 

> How to set up Webhook if they use GoDaddy to trigger real-time updates?

Setting up webhooks for real-time updates on GoDaddy requires using a VPS or dedicated server plan to support a Node.js environment, as shared hosting is insufficient. As far as I understand, you'd have to deploy your application on this environment, then create a webhook endpoint that Prismic can call when content changes. 

> Is there anything else I should be aware of regarding the PROS and CONS of deploying Prismic beside API routes?

It might be worth asking in the [Prismic Community](https://community.prismic.io/) this question, I'm sure they'd give you more of a detailed response!

**Comment by solutionstack99** [comment]
Thank you for the info. I understand this a bit more. 

Vercel costs $20 per month and scales based on site performance and traffic, correct? Would you recommend this package at the start of a SaaS startup?

[a comment by an automated account is not shown]
